avocado/plugins/run.py

```diff
@@ -161,7 +163,17 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
+
+        # Run JobPre plugins
+        pre_post_dispatcher = JobPrePostDispatcher()
+        output.log_plugin_failures(pre_post_dispatcher.load_failures)
+        pre_post_dispatcher.map_method('pre', job_instance)
+
         job_run = job_instance.run()
+
+        # Run JobPost plugins
+        pre_post_dispatcher.map_method('post', job_instance)
+
```

Comment:
Well even thought this does what is needed, I'd prefer having the pre/post job methods as `Job` methods and part of the `run` workflow. It'd fit nicer to what I imagine behind the Job API. Even from this you can see that the pre/post plugin is actually tied to the Job as it requires the Job as an argument. So imagine I want to run multiple jobs at once:

```python
job_a = job.Job(args1)
job_b = job.Job(args2)
job_a.run()
job_b.run()
```
Should in my opinion include the pre/post of each job without the need to instantiate and run the dispatcher. Alternatively I could use custom workflow:

```python
job_a = job.Job(args1)
job_b = job.Job(args2)
job_a.pre_job()
job_a.create_test_suite()
job_b.pre_job()
job_b.create_test_suite()
job_a.run_tests()
job_b.run_tests()
...
```
and all what I'd need is just the `Job` instance, which would be easier to use (and learn).

Anyway yes, they should be the first/last thing in the workflow and eventually users might want to create `pre_avocado` and `post_avocado` hooks which would be executed before anything and after everything, without any relation to jobs/runners/.... But I don't see them as a priority, nor useful for the default workflow.